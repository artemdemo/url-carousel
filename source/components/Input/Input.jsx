import React from 'react';
import classnames from 'classnames';

import './Input.less';

const Input = (props) => {
    const { type = 'text', value, placeholder, disabled, error, onChange } = props;

    const renderErrorText = () => {
        if (typeof error === 'string' && error !== '') {
            return (
                <div className='input-error text-danger'>
                    {error}
                </div>
            );
        }
        return null;
    };

    const containerClass = classnames({
        'has-error': Boolean(error),
    });

    return (
        <div className={containerClass}>
            <input
                className='form-control'
                type={type}
                value={value}
                placeholder={placeholder}
                disabled={disabled}
                onChange={e => onChange && onChange(e.target.value)} />
            {renderErrorText()}
        </div>
    );
};

Input.propTypes = {
    type: React.PropTypes.string,
    value: React.PropTypes.string,
    placeholder: React.PropTypes.string,
    onChange: React.PropTypes.func,
    disabled: React.PropTypes.bool,
    error: React.PropTypes.oneOfType([
        React.PropTypes.bool,
        React.PropTypes.string,
    ]),
};

export default Input;
