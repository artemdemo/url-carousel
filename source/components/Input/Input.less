.input-error {
    padding-left: 5px;
}
