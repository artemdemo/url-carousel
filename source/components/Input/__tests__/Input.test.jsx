import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Input from '../Input';

describe('<Input>', () => {
    it('Render empty', () => {
        const tree = renderer.create(
            <Input />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('With placeholder', () => {
        const tree = renderer.create(
            <Input
                placeholder='Some placeholder' />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Disabled', () => {
        const tree = renderer.create(
            <Input
                placeholder='Some placeholder'
                disabled />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Set value', () => {
        let changed = false;
        const newValue = 'new value';
        const wrapper = mount(
            <Input
                value=''
                onChange={value => changed = value}
                placeholder='Some placeholder' />
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        const input = wrapper.find('input');
        input.simulate('change', {target: {value: 'new value'}});
        expect(changed).toEqual(newValue);
    });

    describe('Error', () => {
        it('boolean', () => {
            const tree = renderer.create(
                <Input
                    placeholder='Some placeholder'
                    error />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('text', () => {
            const tree = renderer.create(
                <Input
                    placeholder='Some placeholder'
                    error='Some error' />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
