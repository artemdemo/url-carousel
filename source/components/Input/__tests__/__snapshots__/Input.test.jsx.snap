// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> Disabled 1`] = `
<div
  className="input-container"
>
  <input
    className="form-control"
    disabled={true}
    onChange={[Function]}
    placeholder="Some placeholder"
    type="text"
    value={undefined}
  />
</div>
`;

exports[`<Input> Error boolean 1`] = `
<div
  className="input-container has-error"
>
  <input
    className="form-control"
    disabled={undefined}
    onChange={[Function]}
    placeholder="Some placeholder"
    type="text"
    value={undefined}
  />
</div>
`;

exports[`<Input> Error text 1`] = `
<div
  className="input-container has-error"
>
  <input
    className="form-control"
    disabled={undefined}
    onChange={[Function]}
    placeholder="Some placeholder"
    type="text"
    value={undefined}
  />
  <div
    className="input-error"
  >
    Some error
  </div>
</div>
`;

exports[`<Input> Render empty 1`] = `
<div
  className="input-container"
>
  <input
    className="form-control"
    disabled={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value={undefined}
  />
</div>
`;

exports[`<Input> Set value 1`] = `
<Input
  onChange={[Function]}
  placeholder="Some placeholder"
  value=""
>
  <div
    className="input-container"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Some placeholder"
      type="text"
      value=""
    />
  </div>
</Input>
`;

exports[`<Input> With placeholder 1`] = `
<div
  className="input-container"
>
  <input
    className="form-control"
    disabled={undefined}
    onChange={[Function]}
    placeholder="Some placeholder"
    type="text"
    value={undefined}
  />
</div>
`;
